src/scrapper/spiders/slot_spider.py

```diff
@@ -149,35 +154,69 @@ def extractSchedule(self, response):
             end_time = datetime.strptime(schedule["end"], date_format)
 
             for teacher in schedule["persons"]:
+                (sigarra_id, name) = self.get_professor_info(teacher)
+
+                if sigarra_id in self.inserted_teacher_ids:
+                    continue
+
+                self.inserted_teacher_ids.add(sigarra_id)
+
                 yield Professor(
-                    id = teacher["sigarra_id"],
-                    professor_acronym = teacher["acronym"],
-                    professor_name = teacher["name"] #.split("-")[1].strip()
+                    id=sigarra_id,
+                    professor_acronym=teacher["acronym"],
+                    professor_name=name
                 )
 
             for current_class in schedule["classes"]:
                 yield Class(
-                    id=current_class["sigarra_id"],
                     name=current_class["name"],
                     course_unit_id=course_unit_id,
                     last_updated=datetime.now()
                 )
 
+                # INFO Since we need to know at runtime the id of the slot so that we can then create SlotProfessor
+                # instances, we are going to be using the same id as the class in order to minimize database lookups
+                # during the runtime of the scrapper
+                current_class_id = get_class_id(
+                    course_unit_id, current_class["name"])
+
                 yield Slot(
-                    id=current_class["sigarra_id"],
+                    id=current_class_id,
                     lesson_type=schedule["typology"]["acronym"],
                     day=self.days[schedule["week_days"][0]],
                     start_time=start_time.hour + (start_time.minute / 60),
                     duration=(end_time - start_time).total_seconds() / 3600,
                     location=schedule["rooms"][0]["name"],
                     is_composed=len(schedule["classes"]) > 0,
                     professor_id=schedule["persons"][0]["sigarra_id"],
-                    class_id=current_class["sigarra_id"],
+                    class_id=current_class_id,
                     last_updated=datetime.now(),
```

Comment:
Just curiosity. The current_class["sigarra_id"] isn't valid? Why looking at db?

Reply:
It is extremely weird and annoying because apparently the `sigarra_id` of a class (e.g.  `3LEIC01`), is not different depending on the course unit.

For example `3LEIC01` of `CPD` was having the same `current_class["id"]` and the same `current_class["sigarra_id"]` as `3LEIC01` of `IA`, which is why I deemed using these ids unfeasible.

Reply:
I notice that even the `id` is also repeated. 
Actually, it makes sense in a way that, a class have different ucs.
Is not a relationship id class-uc, but just class who have multiples ucs.

However, in practical, we understand that as one, since we choose multiple classes.